app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt

```diff
@@ -330,8 +311,9 @@ class BookmarkFragment : LibraryPageFragment<BookmarkNode>(), UserInteractionHan
             }
         }
 
-        viewLifecycleOwner.lifecycleScope.allowUndo(
-            requireView(), message,
+        MainScope().allowUndo(
```

Comment:
Is there any specific reason why we switched to `viewLifecycleOwner.lifecycleScope` ?

Reply:
I guess it's for keeping the undo after the fragment is gone?

Reply:
Yes, it's not to hold a reference to `viewLifecycleOwner.lifecycleScope` in case the user leaves the fragment

Reply:
It's actually was causing [a leak](https://github.com/mozilla-mobile/fenix/pull/24398#issuecomment-1103255859)